File: python/problems_01_25/problem_02.py
def even_fibsum(limit):
    """
    return the sum of all even terms in the Fibonacci
    series where the largest term is less than the limit
    """
    nums = [1,2]
    the_sum = 0
    while nums[-1] < limit:
        if nums[-1] % 2 == 0:
            the_sum += nums[-1]
        nums.append(nums[-1]+nums[-2])

    return the_sum

File: node/problems_01_25/problem_02.js
// return the sum of all the even terms in the 
// Fibonnaci sequence where the largest term in the seq.
// is less than the limit
function EvenFibSum(limit) {
    var sum = 0;
    var a = 1;
    var b = 2;
    while (b < limit) {
        if (b % 2 == 0) {
            sum += b;
        }
        // this is a cool way of setting (b -> a+b) and (a -> b)
        b = [a + b, a = b][0];
    };
    return sum;
};

module.exports.EvenFibSum = EvenFibSum;
